rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java

```diff
@@ -0,0 +1,339 @@
+import uk.gov.hmcts.ccd.definition.store.repository.entity.AccessTypeRolesEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.JurisdictionEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.model.OrganisationProfileIds;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesRoleResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResults;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesResult;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static junit.framework.TestCase.assertFalse;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.MockitoAnnotations.openMocks;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.http.MediaType.TEXT_PLAIN;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(classes = {EntityToResponseDTOMapperImpl.class})
+@ExtendWith(SpringExtension.class)
+public class AccessTypeRolesControllerTest {
+
+    private AccessTypeRolesController controller;
+    @Spy
+    private  EntityToResponseDTOMapper entityToResponseDTOMapper = new EntityToResponseDTOMapperImpl();
+    @MockBean
+    private AccessTypeRolesRepository accessTypeRolesRepository;
+    private MockMvc mockMvc;
+    private final List<String> orgProfileIds = List.of(new String[]{"SOLICITOR_ORG", "SOLICITOR_ORG"});
+    @Mock
+    private JurisdictionEntity jurisdictionEntity = new JurisdictionEntity();
+    @Mock
+    private JurisdictionEntity jurisdictionEntity2 = new JurisdictionEntity();
+    @Mock
+    private  CaseTypeEntity caseTypeEntity = new CaseTypeEntity();
+    @Mock
+    private  CaseTypeEntity caseTypeEntity2 = new CaseTypeEntity();
+    private static final String CASE_TYPE_REFERENCE = "get-test";
+    private static final String JURISDICTION = "BEFTA_MASTER";
+    private static final String JURISDICTION2 = "BEFTA_MASTER2";
+    @Mock
+    private AccessTypeRolesEntity accessTypeRolesEntity;
+    @Mock
+    private AccessTypeRolesEntity accessTypeRolesEntity1;
+    @Mock
+    private AccessTypeRolesEntity accessTypeRolesEntity2;
+    private AccessTypeRolesJurisdictionResults jurisdictionResults;
+    private List<AccessTypeRolesJurisdictionResult> accessTypeRolesJurisdictions;
+
```

Comment:
I think we need a test that verifies that the resulting JSON looks as its meant to.